solutions/cf-1702E.mdx

```diff
@@ -0,0 +1,128 @@
+---
+id: cf-1702E
+source: CF
+title: Split Into Two Sets
+author: Aadit Ambadkar
+---
+
+[Official Editorial (C++)](https://codeforces.com/blog/entry/104763)
+
+## Explanation
+
+**Time Complexity:** $\mathcal{O}(n)$
+
+First observe that if any number is present on more than 2 dominoes or if a dominoe has the same number repeated twice, then 
+it is impossible to create the 2 sets.
+
+We can turn this problem into
+a graph problem by drawing edges between dominoes that share numbers. Specifically, let domino $i$ have the two numbers 
```

Comment:
why would we think of this as a graph problem?
what cues are there?

Reply:
tbh none thats the trick of this problem. the only motivation is that there are 2 sets so we might try bipartite splits, and for that we need a graph.

Reply:
bruhment